spec/unit/lib/email_spec.rb

```diff
@@ -40,18 +40,9 @@
 
   describe '.find_by_recipient' do
     let(:recipient) { 'test@user.com' }
-    subject { described_class.find_by_recipient(recipient) }
+    subject { described_class.find_by_recipient(recipient, message_subject) }
     it do
-      expect(described_class).to receive(:find).with(recipient, message_subject).once
-      subject
-    end
-  end
-
-  describe '.find' do
-    let(:recipient) { 'test@user.com' }
-    subject { described_class.find(recipient, message_subject) }
-    it do
-      expect(described_class.adapter).to receive(:find).with(recipient, message_subject).once
+      expect(described_class).to receive(:find_by_recipient).with(recipient, message_subject).once
```

Comment:
Missing tests which are required to implement:
1) simple subject without parameters
2) complex subject with 1 parameter
3) complex subject with 2 parameters
4) missing subject

+ we need to cover dsl method **subject** in order to check @subject is setup correctly